documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -0,0 +1,517 @@
+configuration based on Hibernate's `ServiceRegistry` architecture, by using a
+`ReactiveServiceRegistryBuilder`. But that's outside the scope of this document.
+
+=== Obtaining a reactive session
+
+Persistence operations are exposed via a reactive `Session` object. It's very
+important to understand that most operations of this interface are non-blocking,
+and execution of SQL against the database is never performed synchronously.
+Persistence operations that belong to a single unit of work must be chained by
+composition within a single reactive stream.
+
+Also remember that a Hibernate session is a lightweight object that should be
+created, used, and then discarded within a single logical unit of work.
+
+TIP: That is to say, you should reuse the same session across multiple
+persistence operations within a single reactive stream representing a certain
+unit of work, but don't share a session between different reactive streams!
+
+To obtain a reactive `Session` from the `SessionFactory`, use `withSession()`:
+
+[java]
+----
+sessionFactory.withSession(
+        session -> ... //do some work
+);
+----
+
+Alternatively, you may use `openSession()`, but you must remember to `close()`
+the session when you're done.
+
+[java]
+----
+sessionFactory.openSession()
+        .thenCompose(
+            session -> ... //do some work
+                    .whenComplete( ($,e) -> session.close() )
+        );
+----
+
+=== Using the reactive session
+
+The `Session` interface has methods with the same names as methods of the
+JPA `EntityManager`. You might already be familiar with the following session
+operations defined by JPA:
+
+- `find()` obtains a persistent object given its id (its primary key),
+- `persist()` makes a transient object persistent and schedules a SQL
+  insert statement,
+- `remove()` makes a persistent object transient and schedules a SQL
+  delete statement,
+- `merge()` copies the state of a transient object to a corresponding
+   persistent instance and returns the persistent object,
+- `refresh()` refreshes the persistent state of an object,
+- `lock()` obtains a pessimistic lock on a persistent object,
+- `flush()` synchronizes the database state with the current session by
+   executing SQL insert, update, and delete statements, and
+- `detach()` detaches a persistent object from a session without affecting
+   the database.
+
+If you're not familiar with these operations, don't despair! Their semantics
+are defined in the JPA specification, and in the API documentation, and are
+explained in innumerable articles and blog posts.
+
+Now, _here's where Hibernate Reactive is different:_ in the reactive API, each
+of these methods returns its result in a non-blocking fashion via a Java
+`CompletionStage` (or Mutiny `Uni`). For example:
+
+[java]
+----
+session1.find(Book.class, book.id)
+        .thenAccept( book -> System.out.println(book.title + " is a great book!") )
+----
```

Comment:
Maybe we should mention that from now on the examples will show the `Stage.Session` API.

Reply:
Meh. I guess.

Reply:
Done.